sensing/preprocessor/pointcloud/pointcloud_preprocessor/src/crop_box_filter/crop_box_filter_nodelet.cpp

```diff
@@ -91,16 +85,45 @@ CropBoxFilterComponent::CropBoxFilterComponent(const rclcpp::NodeOptions & optio
 }
 
 void CropBoxFilterComponent::filter(
-  const PointCloud2ConstPtr & input, const IndicesPtr & indices, PointCloud2 & output)
+  const PointCloud2ConstPtr & input, [[maybe_unused]] const IndicesPtr & indices,
+  PointCloud2 & output)
 {
   boost::mutex::scoped_lock lock(mutex_);
-  pcl::PCLPointCloud2::Ptr pcl_input(new pcl::PCLPointCloud2);
-  pcl_conversions::toPCL(*(input), *(pcl_input));
-  impl_.setInputCloud(pcl_input);
-  impl_.setIndices(indices);
-  pcl::PCLPointCloud2 pcl_output;
-  impl_.filter(pcl_output);
-  pcl_conversions::moveFromPCL(pcl_output, output);
+
+  output.data.clear();
+  output.data.reserve(input->data.size());
+  for (size_t i = 0U; i + input->point_step < input->data.size(); i += input->point_step) {
+    const pcl::PointXYZ * pt = reinterpret_cast<const pcl::PointXYZ *>(&input->data[i]);
```

Comment:
It might be simpler to access it as follows.
```
      for (sensor_msgs::PointCloud2ConstIterator<float> iter_x(input, "x"), iter_y(input, "y"), iter_z(input, "z"); iter_x != iter_x.end(); ++iter_x, ++iter_y, ++iter_z) {
```

Reply:
fix in https://github.com/tier4/AutowareArchitectureProposal.iv/pull/773/commits/50125aa222dd2786a3c638e463cd6462897c9a86